content/blog/hackaton-2022-building-a-pong.md

```diff
@@ -0,0 +1,212 @@
+
+Our team decided to put the WDA plugin functionnality to good use and we decided to build a pong game that would be playable between two callers using WDA. 
+
+To accomplish that feat we used existing tools from the platform, webrtc, asterisk, nginx, etc and some that are not yet available i.e. the plugin interface that is in the works WDA. 
+
+We started by finding a pong javascript [codebase](https://gist.github.com/straker/81b59eecf70da93af396f963596dfdc5) that was functional and had all the feature we needed. We used an open-source project and we built on that. 
+
+The issues we faced were mostly related to synchronisation and control of the paddles for the players. 
+
+## Intercommunication 
+
+Since we used DTMF's to control the paddles ( we wanted it to work with a plastic phone too ) we had to use the wazo-sdk javascript library. We linked the user control ( arrow up and arrow down ) to send DTMF's to the stack. 
+
+```javascript
+/**
+* When using the wazo-sdk we have directly acces to Wazo websocket events 
+* without too much work. 
+*/
+Wazo.Websocket.on('call_dtmf_created', ({ data }) => {
+
+	if (!data.user_uuid) {
+		return;
+	}
+
+	if (data.digit === '0' || data.digit === '1') {
+		const isPlayer1 = data.digit === '0';
+		return stopMovePlayer(isPlayer1);
+	}
+	/**
+	* Since we need to separate both player we sent different DTMF's
+	* Through the system.
+	*/
+	const isUp = data.digit === '8' || data.digit === '9';
+	const isPlayer1 = data.digit === '2' || data.digit === '8';
+	
+	movePlayer(isPlayer1, isUp);
+});
+
+```
+
+We also had to decide how to synchronise the ball between the two users, since we are not using a netcode library, we decided to make the caller the source of truth. One of the interesting part is that we are able to speak while playing the game. Oh and if you are very good at pong your calls wont be very long, we needed a way to make a winner so when you get to 20 points the game ends and the call hangs-up. 
+```javascript
+setOnGameEndedCallback(idPlayer => {
+	Wazo.Phone.sendMessage(JSON.stringify({ type: 'playerWon', idPlayer }), call.sipSession);
+	onPlayerWon(idPlayer);
+	Wazo.Phone.hangup(call);
+});
+```
+
+## The Deployment
+
+Let's make it (not) production ready:
+
+### Industrialize deployment
+We use ansible to deploy this quickly. Fortunately we already have an nginx service running.
+First of all, let's create the ansible project
+```
+ansible-galaxy init ansible-role-hackapong
+cd ansible-role-hackapong
+```
+Let's take a minute to think about it. 
+We need to **(1)** download a project from git, **(2)** upload the nginx configuration and **(3)** enable this new site.
+
+We fill the variables first.
+Are needed the different values.
+For **(1)**
+- The code repository
+- Path of the cloned project=> /var/www/hackapong
+
+For **(2)** & **(3)**
+- Nginx sites-available project name => hackapong
+- A FQDN for the incoming requests => hackapong.wazo.io
+
+The arising variable file :
```

Comment:
Arising?

Reply:
I dont know i will ask  Alex hoguin

Reply:
Changed the wording to:
The ensuing variable file